### PR TITLE
Feat/ecs

### PR DESCRIPTION
**Pull request by Lelong-vincent**

### CODE REVIEW

**Comment by kevinpolossat** [comment]
could you please remove .dll and restore/readd the previous gitignore ?

File: src/AGameState.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#include "AWorld.h"
+
+class GameEngine;
+
+class AGameState {
+public:
+	AGameState() = default;
+	AGameState(AGameState const & other) = delete;
+	AGameState(AGameState const && other) = delete;
+	virtual ~AGameState() = default;
```
Comment (kevinpolossat) [inline src/AGameState.h]: move constructor should not be const => AGameState(AGameState && other)

Comment (kevinpolossat) [reply src/AGameState.h]: How do you move something const : O

File: src/AWorld.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+#include <bitset>
+#include <array>
+#include <SFML/Graphics.hpp>
+#include "Components.h"
+
+class AWorld {
+public:
+	AWorld() = default;
+	AWorld(AWorld const & other) = delete;
+	AWorld(AWorld const && other) = delete;
```
Comment (kevinpolossat) [inline src/AWorld.h]: same here for move operator

File: src/GameEngine.h [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+using PrioritizedDrawable = std::pair<int32_t , std::shared_ptr<sf::Drawable>>;
+
+class GameEngine {
+public:
+	static constexpr int32_t UI = 0;
+	static constexpr int32_t Background = -1;
+
+	GameEngine();
+	GameEngine(GameEngine const & other) = delete;
+	GameEngine(GameEngine const && other) = delete;
```
Comment (kevinpolossat) [inline src/GameEngine.h]: same here for move constructor

File: src/IntroState.h [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+class IntroState : public AGameState {
+	enum ButtonId {
+		START,
+		QUIT
+	};
+
+public:
+	IntroState() = default;
+	IntroState(IntroState const & other) = delete;
+	IntroState(IntroState const && other) = delete;
```
Comment (kevinpolossat) [inline src/IntroState.h]: ^

File: src/IntroWorld.h [hunk 5]
```diff
@@ -0,0 +1,28 @@
+#ifndef WORLD_H_
+#define WORLD_H_
+
+#include "AWorld.h"
+
+class IntroWorld : public AWorld {
+public:
+	IntroWorld() = default;
+	IntroWorld(IntroWorld const & other) = delete;
+	IntroWorld(IntroWorld const && other) = delete;
```
Comment (kevinpolossat) [inline src/IntroWorld.h]: ^

File: src/IntroWorld.h [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+	~IntroWorld() override = default;
+
+	IntroWorld & operator=(IntroWorld const & other) = delete;
+	IntroWorld & operator=(IntroWorld const && other) = delete;
+
```
Comment (kevinpolossat) [inline src/IntroWorld.h]: ^

File: src/IntroState.h [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+	~IntroState() override = default;
+
+	IntroState & operator=(IntroState const & other) = delete;
+	IntroState & operator=(IntroState const && other) = delete;
+
```
Comment (kevinpolossat) [inline src/IntroState.h]: ^

File: src/PlayState.h [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+#include "AGameState.h"
+#include "PlayWorld.h"
+
+class PlayState : public AGameState {
+public:
+	PlayState() = default;
+	PlayState(PlayState const & other) = delete;
+	PlayState(PlayState const && other) = delete;
+	~PlayState() override = default;
```
Comment (kevinpolossat) [inline src/PlayState.h]: ^

File: src/PlayState.h [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+	PlayState & operator=(PlayState const & other) = delete;
+	PlayState & operator=(PlayState const && other) = delete;
+
```
Comment (kevinpolossat) [inline src/PlayState.h]: ^

File: src/PlayWorld.h [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+#define PLAYWORLD_H_
+
+#include "AWorld.h"
+
+class PlayWorld : public AWorld {
+public:
+	PlayWorld() = default;
+	PlayWorld(PlayWorld const & other) = delete;
+	PlayWorld(PlayWorld const && other) = delete;
+	~PlayWorld() override = default;
```
Comment (kevinpolossat) [inline src/PlayWorld.h]: ^

File: src/PlayWorld.h [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+	PlayWorld & operator=(PlayWorld const & other) = delete;
+	PlayWorld & operator=(PlayWorld const && other) = delete;
+
```
Comment (kevinpolossat) [inline src/PlayWorld.h]: ^

File: src/ResourcesManager.h [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+
+#include <unordered_map>
+#include <SFML/Graphics.hpp>
+
+class ResourcesManager {
+public:
+	ResourcesManager() = default;
+	ResourcesManager(ResourcesManager const & other) = delete;
+	ResourcesManager(ResourcesManager const && other) = delete;
+	~ResourcesManager() = default;
```
Comment (kevinpolossat) [inline src/ResourcesManager.h]: ^

File: src/ResourcesManager.h [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+
+	ResourcesManager & operator=(ResourcesManager const & other) = delete;
+	ResourcesManager & operator=(ResourcesManager const && other) = delete;
+
```
Comment (kevinpolossat) [inline src/ResourcesManager.h]: ^

File: src/AWorld.h [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+	virtual ~AWorld() = default;
+
+	AWorld & operator=(AWorld const & other) = delete;
+	AWorld & operator=(AWorld const && other) = delete;
+
```
Comment (kevinpolossat) [inline src/AWorld.h]: ^

File: src/GameEngine.h [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+	~GameEngine();
+
+	GameEngine & operator=(GameEngine const & other) = delete;
+	GameEngine & operator=(GameEngine const && other) = delete;
+
```
Comment (kevinpolossat) [inline src/GameEngine.h]: ^

**Review by kevinpolossat — CHANGES_REQUESTED** [review]

File: src/AGameState.h (hunk 1 above)
Comment (Lelong-vincent) [reply src/AGameState.h]: Done for al

**Review by kevinpolossat — APPROVED** [review]

**Pull request merged**